client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamFactoryImpl.java

```diff
@@ -53,8 +53,8 @@ public EventSegmentReader createEventReaderForSegment(Segment segment, int buffe
     }
 
     @Override
-    public EventSegmentReader createEventReaderForSegment(Segment segment, long startOffset, int bufferSize) {
-        return getEventSegmentReader(segment, null, startOffset, Long.MAX_VALUE, bufferSize);
+    public EventSegmentReader createEventReaderForSegment(Segment segment, long startOffset, int lengthToRead) {
+        return getEventSegmentReader(segment, null, startOffset, startOffset + lengthToRead, lengthToRead);
     }
 
     private EventSegmentReader getEventSegmentReader(Segment segment, Semaphore hasData, long startOffset, long endOffset, int bufferSize) {
```

Comment:
In this file **bufferSize** param can be changed to **lengthToRead** to maintain uniformity across the code base.